docs/api.rst

```diff
@@ -5,6 +5,8 @@
     Distributed under the Boost Software License, Version 1.0. (See accompanying
     file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
+:tocdepth: 3
+
```

Comment:
I'm setting this explicitly to avoid having _everything_ show up in the table of contents on the right hand side.

- level 1 is basically only the page title "API reference"
- level 2 adds the header sections (`pika/execution.hpp` etc.)
- level 3 adds the functions/classes in each header
- level 4 adds member functions of classes (or parameters of functions, only with sphinx-immaterial)
- level 5 adds pre/post conditions of member functions (only with sphinx-immaterial)

I'm leaning towards 3 or 4 with a slight preference for 3, which is what I've currently chosen.
-

Reply:
Stick to level 3, do not add too much information in table of contents.